NebulaHost/PacketProcessors/Logistics/ILSgStationPoolSyncRequestProcessor.cs

```diff
@@ -0,0 +1,149 @@
+﻿using NebulaModel.Attributes;
+using NebulaModel.Networking;
+using NebulaModel.Packets.Logistics;
+using NebulaModel.Packets.Processors;
+using NebulaModel.DataStructures;
+
+namespace NebulaHost.PacketProcessors.Logistics
+{
+    [RegisterPacketProcessor]
+    public class ILSgStationPoolSyncRequestProcessor : IPacketProcessor<ILSRequestgStationPoolSync>
+    {
+        private PlayerManager playerManager;
+        public ILSgStationPoolSyncRequestProcessor()
+        {
+            playerManager = MultiplayerHostSession.Instance.PlayerManager;
+        }
+        public void ProcessPacket(ILSRequestgStationPoolSync packet, NebulaConnection conn)
+        {
+            Player player = playerManager.GetPlayer(conn);
+            if(player == null)
+            {
+                player = playerManager.GetSyncingPlayer(conn);
+            }
+            if (player != null)
+            {
+                int countILS = 0;
+                int iter = 0;
+
+                foreach(StationComponent stationComponent in GameMain.data.galacticTransport.stationPool)
+                {
+                    if(stationComponent != null)
+                    {
+                        countILS++;
+                    }
+                }
+
+                if(countILS == 0)
+                {
+                    return;
+                }
+
+                int[] stationGId = new int[countILS];
+                int[] stationId = new int[countILS];
+                Float3[] DockPos = new Float3[countILS];
+                Float4[] DockRot = new Float4[countILS];
+                int[] planetId = new int[countILS];
+                int[] workShipCount = new int[countILS];
+                int[] idleShipCount = new int[countILS];
+                ulong[] workShipIndices = new ulong[countILS];
+                ulong[] idleShipIndices = new ulong[countILS];
+
+                int[] shipStationGId = new int[countILS * 10];
```

Comment:
I feel like a const as opposed to a fixed "10" would work better here, in the event someone wants compatibility with a mod that increases capacities, or the game has ILSes tweaked in the future - I see it repeated further down also